enter.c

```diff
@@ -161,15 +161,55 @@ int enter(struct entry_settings *opts)
 		errx(1, "cannot create NICs when not in a network namespace");
 	}
 
+
+	char cgroup_path[BUFSIZ];
+	/*
+	 * If there is a cgroup specified we need to create bst.<pid> subcgroup to join
+	 * Otherwise if no cgroup is specified but limits are applied bst will error out
+	 */
+	if (opts->nactiveclimits != 0 || opts->cgroup_path != NULL) {
+		// If cgroup path is unspecified then attempt to derive from /proc/self/cgroup
+		if (opts->cgroup_path == NULL) {
+			FILE *selfcgroupfd = fopen("/proc/self/cgroup", "r");
+			if (selfcgroupfd == NULL) {
+				err(1, "unable to derive current cgroup hierarchy from /proc/self/cgroup");
+			}
+
+			char selfcgroup[BUFSIZ];
+
+			while (fgets(selfcgroup, BUFSIZ, selfcgroupfd) != NULL) {
+				if (strncmp(selfcgroup, "0::/", 3) == 0) {
```

Comment:
the size argument for strncmp is supposed to be the size of the buffer, not the size of the string constant; this should be BUFSIZ, even if it seems a bit ridiculous.

The reason why is that if for some insane reason, BUFSIZ is 2, then this ends up reading past the end of the selfcgroup buffer, whereas the string constant is always well-formed, and strncmp stops as soon as its sees the null terminator of the string constant.